rebasebot/bot.py

```diff
@@ -76,7 +76,13 @@ def _commit_go_mod_updates(gitwd: git.Repo, source: GitHubBranch) -> None:
                 full_path = os.path.join(module_base_path, filename)
                 if not os.path.exists(full_path):
                     continue
-                gitwd.remotes.source.repo.git.checkout(f"source/{source.branch}", full_path)
+                try:
+                    gitwd.remotes.source.repo.git.checkout(f"source/{source.branch}", full_path)
+                except Exception as err:
```

Comment:
I think you catch specific types of exceptions rather than all exceptions for best practice

Reply:
That's correct; it's better if you can catch a specific list. This is slightly different than Go, in that Go often generically returns any error caught (`if err != nil`).

Reply:
I think you want [CheckoutException](https://github.com/gitpython-developers/GitPython/blob/f55d194b01d980c28c8ce966c9f95daf868ed8ec/git/exc.py#L156), which I think is exposed via `git.CheckoutException` through a bunch of imports statements.

Reply:
Yeah that's a good idea. I've update the PR to adopt this.